packages/config/config/devices/0x014a/tstat-tbz500.json

```diff
@@ -0,0 +1,712 @@
+	"devices": [
+		{
+			"productType": "0x5442",
+			"productId": "0x5439"
+		}
+	],
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"paramInformation": [
+		{
+			"#": "1",
+			"label": "System Type",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 1,
+			"defaultValue": 0,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Standard",
+					"value": 0
+				},
+				{
+					"label": "Heat Pump",
+					"value": 1
+				}
+			]
+		},
+		{
+			"#": "2",
+			"label": "Fan Type",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 1,
+			"defaultValue": 0,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Gas (No fan w/Heat)",
+					"value": 0
+				},
+				{
+					"label": "Electric (Fan w/Heat)",
+					"value": 1
+				}
+			]
+		},
+		{
+			"#": "3",
+			"label": "Changeover Type",
+			"description": "For Heat Pump HVAC systems only",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 1,
+			"defaultValue": 0,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Changeover with Cooling systems (Orange wire)",
+					"value": 0
+				},
+				{
+					"label": "Changeover with Heating systems (Brown wire)",
+					"value": 1
+				}
+			]
+		},
+		{
+			"#": "4",
+			"label": "2nd Stage Heat",
+			"$import": "~/templates/master_template.json#base_enable_disable"
+		},
+		{
+			"#": "5",
+			"label": "Aux Heat",
+			"$import": "~/templates/master_template.json#base_enable_disable",
+			"defaultValue": 1
+		},
+		{
+			"#": "6",
+			"label": "2nd Stage Cool",
+			"$import": "~/templates/master_template.json#base_enable_disable"
+		},
+		{
+			"#": "7",
+			"label": "Temperature Unit",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 1,
+			"defaultValue": 1,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Celcius",
+					"value": 0
+				},
+				{
+					"label": "Fahrenheit",
+					"value": 1
+				}
+			]
+		},
+		{
+			"#": "8",
+			"label": "Minimum Off Time",
+			"unit": "minutes",
+			"valueSize": 1,
+			"minValue": 5,
+			"maxValue": 9,
+			"defaultValue": 5
+		},
+		{
+			"#": "9",
+			"label": "Minimum Run Time",
+			"unit": "minutes",
+			"valueSize": 1,
+			"minValue": 3,
+			"maxValue": 9,
+			"defaultValue": 3
+		},
+		{
+			"#": "10",
+			"label": "Setpoint Heat/Cool Delta",
+			"description": "Sets the minimum separation between heating and cooling setpoints.",
+			"unit": "°F",
+			"valueSize": 1,
+			"minValue": 3,
+			"maxValue": 15,
+			"defaultValue": 3
+		},
+		{
+			"#": "11",
+			"label": "Heating Stage 1 On Threshold",
+			"description": "Sets the delta from setpoint that stage 1 heating starts.",
+			"unit": "°F",
+			"valueSize": 1,
+			"minValue": 1,
+			"maxValue": 6,
+			"defaultValue": 1
+		},
+		{
+			"#": "12",
+			"label": "Heating Stage 1 Off Threshold",
+			"description": "Sets the delta from setpoint that stage 1 heating stops. Stage 1 turns off at setpoint + Delta Stage 1.",
```

Comment:
I'm a bit confused by this description. Does "Delta Stage 1" refer to this parameter or no. 10?

Reply:
Neither. It actually refers to No. 11. 

No. 11 is a delta from the Setpoint that heat turns ON.  No. 12 is a delta from No. 11 that the Heat turns OFF (= Setpoint + No. 11). I changed the text to match the verbiage in the User Guide in the Setup table. This is what MOST users will use to configure the unit on the device itself. It is more user friendly than the Parameter verbiage for the same settings (which is what I had).

I guess the larger question for me is, are these supposed to be stand alone descriptions that can be understandable in the absence of a user manual? I think that would require a lot more information, unless one is an HVAC expert. I approached it as as "ok, you've read the manual, set up your device on its screen, and now you want to do some more zwave-based fiddling" type of language, and "here's some hints as to what you're changing".

I guess we could replace "Delta Stage 1" with "the Delta from Parameter 11"?

Reply:
I know that many users rely on the defined parameters alone without checking the manual, except for more complicated cases (which this one might be).

I checked the manual and I understand this a bit differently than you just described it. For me it seems like:
Stage 1 starts heating at **Setpoint + Param 11** and stop heating at **Setpoint + Param 12** (which is lower)
Stage 2 starts heating at **Setpoint + Param 13** and stop heating at **Setpoint + Param 14**
etc. (negative for the cooling stages).

If that makes sense, I would recommend this:
**Param 11**:
label: "Heating Stage 1: On Delta"
description: "Stage 1 starts heating at _\<Setpoint> + \<Heating Stage 1: On Delta>_"

**Param 12**:
label: "Heating Stage 1: Off Delta"
description: "Stage 1 stops heating at _\<Setpoint> + \<Heating Stage 1: Off Delta>_"

and so on. What do you think? This way, the params would be self-explanatory without having to refer to the manual or paramter numbers.

Reply:
You are correct! I was wrong in my description (and understanding). After further reading of other thermostat manuals, in particular the RCS TZ45 (which is basically THIS one), I found the following statement:

> Note on Delta Settings : The Delta T Setting is the delta, or difference between, the Setpoint and
current temp for determining when a heat or cool call comes on. The “delta” is the number of
degrees away from Setpoint .

I'm not sure what you mean when you say: "Param 12 (which is lower)". It is truly a 'delta' in that it has no +/-, i.e., direction is inferred by the mode. To restate what I think it means (which I believe is what you said):

Heat:
Stage 1 starts when temp is No. 11 deg below Setpoint and runs until temp is EITHER No. 12 deg below Setpoint (goes off) OR temp is No. 13 deg below Setpoint , at which point Stage 2 starts and continues until temp is EITHER No. 14 deg below Setpoint (goes off) OR temp is No. 15 deg below Setpoint , at which point Stage 3 starts and continues until temp is No 16 deg below Setpoint . 

Cool:
Stage 1 starts when temp is No. 17 above Setpoint and runs until temp is EITHER No. 18 deg above Setpoint  (goes off) OR temp is No. 19 deg above Setpoint, at which point Stage 2 starts and continues until temp is No. 20 deg above Setpoint.

What about this?
Param 11:
label: "Heating Stage 1: On Delta"
description: "Stage 1 starts heating when current temp is <Heating Stage 1: On Delta> deg below <Setpoint>"

Param 12:
label: "Heating Stage 1: Off Delta"
description: "Stage 1 stops heating when current temp is <Heating Stage 1: Off Delta> deg below <Setpoint>" 

Param 13:
label: "Heating Stage 2: On Delta"
description: "Stage 2 starts heating when current temp is <Heating Stage 2: On Delta> deg below <Setpoint>"

Param 14:
label: "Heating Stage 2: Off Delta"
description: "Stage 2 stops heating when current temp is <Heating Stage 2: Off Delta> deg below <Setpoint>" 

Param 15:
label: "Heating Stage 3: On Delta"
description: "Stage 3 starts heating when current temp is <Heating Stage 3: On Delta> deg below <Setpoint>"

Param 16:
label: "Heating Stage 3: Off Delta"
description: "Stage 3 stops heating when current temp is <Heating Stage 3: Off Delta> deg below <Setpoint>" 

Param 17:
label: "Cooling Stage 1: On Delta"
description: "Stage 1 starts cooling when current temp is <Cooling Stage 1: On Delta> deg above <Setpoint>"

Param 18:
label: "Cooling Stage 1: Off Delta"
description: "Stage 1 stops cooling when current temp is <Cooling Stage 1: Off Delta> deg above <Setpoint>" 

Param 18:
label: "Cooling Stage 2: On Delta"
description: "Stage 2 starts cooling when current temp is <Cooling Stage 2: On Delta> deg above <Setpoint>"

Param 20:
label: "Cooling Stage 2: Off Delta"
description: "Stage 2 stops cooling when current temp is <Cooling Stage 2: Off Delta> deg above <Setpoint>" 

Is that your understanding too? We could simplify some by replacing the <X Stage X: On Delta> blocks, such as:

description: "Stage 2 stops cooling when current temperature is this parameter degrees above <Setpoint>" 

Thoughts? The "+" sign used in the manual has always been confusing to me.

Reply:
> I'm not sure what you mean when you say: "Param 12 (which is lower)"

I meant the range (0-6) is smaller than for the "ON" part (1-7), so the OFF point is closer to the setpoint than the ON point.

Doesn't matter though, I agree with your parameter descriptions. Lets use the versions that mention the specific labels.
The manual seems to explain the deltas the exact opposite way but that makes no sense IMO.

Reply:
Will do. Thanks for your patience!